libffi 3.3 deprecates ffi_prep_closure

In libffi 3.3, one gets this warning:
```
./foreign.c: In function ‘ffi_callback_or_curry’:
./foreign.c:4386:3: warning: ‘ffi_prep_closure’ is deprecated: use ffi_prep_closure_loc instead [-Wdeprecated-declarations]
 4386 |   if (ffi_prep_closure(cl, cif, do_callback, (void*)(cl_cif_args->data))
      |   ^~
In file included from ./foreign.c:86:
/usr/lib64/libffi-3.3_rc0/include/ffi.h:332:1: note: declared here
  332 | ffi_prep_closure (ffi_closure*,
      | ^~~~~~~~~~~~~~~~
```

We should start using `ffi_prep_closure_loc` if available.

--- a/racket/src/foreign/foreign.c
+++ b/racket/src/foreign/foreign.c
@@ -4383,7 +4383,7 @@ static Scheme_Object *ffi_callback_or_curry(const char *who, int curry, int argc
   }
 # endif /* MZ_USE_MZRT */
   cl_cif_args->data = callback_data;
-  if (ffi_prep_closure(cl, cif, do_callback, (void*)(cl_cif_args->data))
+  if (ffi_prep_closure_loc(cl, cif, do_callback, (void*)(cl_cif_args->data), cl)
       != FFI_OK)
     scheme_signal_error
       ("internal error: ffi_prep_closure did not return FFI_OK");

